### ISSUE TITLE
The portfolio does not freeze holdings while a limit order is submitted (pending)

### ISSUE THREAD

**Issue by BadgerTheAdger** [QuantConnect/Lean#4987]
#### Expected Behavior
A limit order is placed, the amounts of the currency being sold have been frozen/subtracted. If an order cancels, the amounts of that currency unfreeze. 

#### Actual Behavior
The amount of the currency being sold does not freeze.

#### Potential Solution
Perhaps modify the order event handler logic in such a way, that it subtracts submitted funds when the order is submitted. 
I have found a similar issue #2161 however that's for market orders. For Limit orders, I have the same issue.

What do you guys think, is it an issue at all? Or it's good behavior? I'm new to trading, I'm simply implementing my own GUI client and I see that it might not be the best in terms of UX. 

#### Checklist
- [X] I have completely filled out this template
- [X] I have confirmed that this issue exists on the current `master` branch
- [X] I have confirmed that this is not a duplicate issue by searching [issues](https://github.com/QuantConnect/Lean/issues)

**Comment by Martin-Molinero** [comment]
Hey Mark!
`currency being sold have been frozen/subtracted` not sure this is a bug/expected behavior. I think what you want is something like `Transactions.GetOpenOrders()`? Not sure what you use case looks like from the algorithm perspective

**Comment by BadgerTheAdger** [comment]
My use case is like this:

1. Get the current holdings via Algorithm.Portfolio
2. Sell 95% of them

But if I already placed a limit order for 50% of holdings, and it's pending, my available portfolio is now 50%. But my algorithm still thinks it's 100% and so it places 95% of 100% instead of 95% of 50%. And I get 'insufficient available balance' on an exchange

**Comment by pomeara** [comment]
@Martin-Molinero I agree with @BadgerTheAdger if an order is placed and it's in the 'pending state' the funds are locked and should not be available to attempt to place any further trades as funds are locked when a trade is placed and it seems lean does not respect this situation

**Comment by pomeara** [comment]
@Martin-Molinero have you perhaps any new ideas about this item.   If you place a trade the funds are 'frozen' on an exchange typically and then you cant 'double spend' any money - I don't think this behavior change between different stock exchanges/crypto exchanges ?
